packages/compiler-sfc/__tests__/compileStyle.spec.ts

```diff
@@ -181,12 +181,40 @@ color: red
       ".foo .bar { color: red;
       }"
     `)
-    // global ignores anything before it
+
     expect(compileScoped(`.baz .qux ::v-global(.foo .bar) { color: red; }`))
       .toMatchInlineSnapshot(`
-      ".foo .bar { color: red;
+      ".baz .qux[data-v-test] .foo .bar { color: red;
```

Comment:
The original behavior was that `global` would ignore the preceding rules, but it will no longer do so after this PR. Not sure if this will introduce some edge cases.